weather.py

```diff
@@ -0,0 +1,103 @@
+from abc import ABC, abstractmethod
+
+class WeatherService(ABC):
+    """
+    Abstract base class for weather services. Defines the interface for fetching and parsing weather data.
+    """
+
+    @abstractmethod
+    def fetching_data(self, city: str) -> dict:
+        """
+        Fetch weather data for the given city.
+
+        :param city: Name of the city to fetch weather data for.
+        :return: Dictionary containing the fetched weather data.
+        """
+        pass
+
+    @abstractmethod
+    def temperature(self, data: dict) -> float:
+        """
+        Parse and return the temperature from the fetched data.
+
+        :param data: Dictionary containing the fetched weather data.
+        :return: Temperature as a float.
+        """
+        pass
+
+    @abstractmethod
+    def humidity(self, data: dict) -> float:
+        """
+        Parse and return the humidity from the fetched data.
+
+        :param data: Dictionary containing the fetched weather data.
+        :return: Humidity as a float.
+        """
+        pass
+
+class WebWeatherService(WeatherService):
```

Comment:
Maybe rename `WebWeatherService` to something more specific, since "Web" is a bit broad, i.e. "API", or maybe even the specific api you use (i.e. OpenMeteo)